update-common/src/artifacts/update_plan.rs

```diff
@@ -700,38 +741,21 @@ impl<'a> UpdatePlanBuilder<'a> {
             }
         }
 
-        // Ensure that all A/B RoT images for each board kind have the same
-        // version number.
-        for (kind, mut single_board_rot_artifacts) in [
-            (
-                KnownArtifactKind::GimletRot,
-                self.gimlet_rot_a.iter().chain(&self.gimlet_rot_b),
-            ),
-            (
-                KnownArtifactKind::PscRot,
-                self.psc_rot_a.iter().chain(&self.psc_rot_b),
-            ),
-            (
-                KnownArtifactKind::SwitchRot,
-                self.sidecar_rot_a.iter().chain(&self.sidecar_rot_b),
-            ),
-        ] {
-            // We know each of these iterators has at least 2 elements (one from
-            // the A artifacts and one from the B artifacts, checked above) so
-            // we can safely unwrap the first.
-            let version =
-                &single_board_rot_artifacts.next().unwrap().id.version;
-            for artifact in single_board_rot_artifacts {
-                if artifact.id.version != *version {
+        // Ensure that all A/B RoT images for each board kind and same
+        // signing key have the same version.
+        for ((kind, _), versions) in self.by_sign {
+            let version = &versions.first().unwrap().version;
```

Comment:
Can you add a comment noting why this `.unwrap()` is safe? (The guarantee that there is at least one element happens pretty far away from this up in `add_rot_artifact`, right?)